src/bitcoinrpcclient/bitcoinrpcclient.go

```diff
@@ -19,7 +19,13 @@ type BitcoinRPCClient struct {
 // NewBitcoinRPCClient returns a new Bitcoin Core RPC Client. This functions
 // waits for a maximum of 10 seconds for the corresponding RPC server to be
 // ready. Otherwise it returns a timeout.
-func NewBitcoinRPCClient(rpcUser string, rpcPass, rpcHost string, rpcPort string) (*BitcoinRPCClient, error) {
+func NewBitcoinRPCClient(rpcUser, rpcPass, rpcHost, rpcPort string) (*BitcoinRPCClient, error) {
+	if len(rpcUser) == 0 || len(rpcPass) == 0 || len(rpcHost) == 0 || len(rpcPort) == 0 {
+		return nil, fmt.Errorf(
+			"received empty parameter (rpcUser=%q rpcPass=%q rpcHost=%q rpcPort=%q)",
```

Comment:
Could leak the RPC user and password into the log. 

(TIL `%q`, thanks!)

Reply:
ah I thought this was test-only

will fix

Reply:
It's currently used in tests only, but I guess we'll use it as soon as we need information over RPC.